conda/recipes/pynvjitlink/meta.yaml

```
@@ -0,0 +1,69 @@
+# Copyright (c) 2018-2023, NVIDIA CORPORATION.
+
+{% set py_version = environ['CONDA_PY'] %}
+{% set cuda_version = '.'.join(environ['RAPIDS_CUDA_VERSION'].split('.')[:2]) %}
+{% set cuda_major = cuda_version.split('.')[0] %}
+{% set date_string = environ['RAPIDS_DATE_STRING'] %}
+
+{% set data = load_file_data("pyproject.toml") %}
+
+package:
+  name: pynvjitlink
+  version: {{ data.get("project", {}).get("version", "") }}
+
+source:
+  path: ../../..
+
+build:
+  noarch: python
+  script:
+    - {{ PYTHON }} -m pip install . -vv --config-settings=cmake.args=-DCUDAToolkit_ROOT=/usr/local/cuda
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.26.4
+    - ninja
+    - sysroot_{{ target_platform }} 2.17
+  host:
+    # We get libnvjitlink from the host's RPM packages, which are updated
+    # sooner than conda sources.
+    #- libnvjitlink-dev
+    #- libnvjitlink-static
+    #- cuda-version ={{ cuda_version }}
+    - python
+    - pip
+    - scikit-build-core
+  run:
+    - python
+    - numba >=0.57
+    - {{ pin_compatible('cuda-version', max_pin='x', min_pin='x') }}
```

If we did build with CUDA 12.2 and wanted to make sure this didn't get installed with CUDA 12.3, we could add an upper bound to the `cuda-version` installed. Perhaps like this?

```suggestion
    - {{ pin_compatible('cuda-version', min_pin='x', max_pin='12.3') }}
```

---

Good question but I'm not sure if we need constraints at runtime. The issue is needing to build with nvjitlink from CUDA 12.3 due to bugs in earlier versions. https://github.com/rapidsai/pynvjitlink/pull/33#discussion_r1445510368